localstack/services/kms/models.py

```diff
@@ -543,8 +545,12 @@ class KmsGrant:
     # simplicity.
     token: str
 
-    def __init__(self, create_grant_request: CreateGrantRequest):
+    def __init__(self, create_grant_request: CreateGrantRequest, account_id: str, region_name: str):
         self.metadata = dict(create_grant_request)
+
+        if not is_valid_key_arn(self.metadata["KeyId"]):
+            self.metadata["KeyId"] = kms_key_arn(self.metadata["KeyId"], account_id, region_name)
```

Comment:
This is changed to always store the KMS key ARN rather than just the ID. 

This is required for cross-account key retiring.